main/acle.md

```diff
@@ -1072,6 +1083,31 @@ including the header:
 
 Including `<arm_sme.h>` also includes [`<arm_sve.h>`](#arm_sve.h).
 
+### Predefined feature macros and header files
+
+Testing a feature macro returns the compiler's availability for the feature via ACLE
+documented intrinsics and inline assembly. This property can be independent of
+where the test is performed, meaning a feature macro does not guarantee the
+validity of using the feature at the point it is tested. For example:
+
+``` c
+    #include <arm_sme.h>
+
+    void foo(svbool_t pg, void *ptr, uint32_t slice_base) {
+    #ifdef __ARM_FEATURE_SME
+      svst1_hor_za8(0, slice_base, pg, ptr);
+    #endif
+    }
+```
+
+Whilst testing `__ARM_FEATURE_SME` ensures the compiler has available the SME
+intrinsic `svst1_hor_za8`, `foo` will fail to compile because it has not been
+decorated with the necessary keywords to guarantee the function will be
+executed in streaming mode.
```

Comment:
```suggestion
While testing `__ARM_FEATURE_SME` ensures that the SME intrinsic `svst1_hor_za8`
is available, `foo` will fail to compile because the call does not occur in a
[streaming statement](#streaming-statement).
```